internal/kibana/role.go

```diff
@@ -370,6 +436,66 @@ func expandKibanaRoleElasticsearch(v interface{}) *kbapi.KibanaRoleElasticsearch
 				elasticConfig.Indices = indices
 			}
 
+			if v, ok := userElasticConfig["remote_indices"]; ok {
+				definedRemoteIndices := v.(*schema.Set)
+				remote_indices := make([]kbapi.KibanaRoleElasticsearchRemoteIndice, definedRemoteIndices.Len())
+				for i, idx := range definedRemoteIndices.List() {
+					index := idx.(map[string]interface{})
+
+					definedNames := index["names"].(*schema.Set)
+					names := make([]string, definedNames.Len())
+					for i, name := range definedNames.List() {
+						names[i] = name.(string)
+					}
+					definedClusters := index["clusters"].(*schema.Set)
+					clusters := make([]string, definedClusters.Len())
+					for i, cluster := range definedClusters.List() {
+						clusters[i] = cluster.(string)
+					}
+					definedPrivileges := index["privileges"].(*schema.Set)
+					privileges := make([]string, definedPrivileges.Len())
+					for i, pr := range definedPrivileges.List() {
+						privileges[i] = pr.(string)
+					}
+
+					newRemoteIndex := kbapi.KibanaRoleElasticsearchRemoteIndice{
+						Names:      names,
+						Clusters:   clusters,
+						Privileges: privileges,
+					}
+
+					if query := index["query"].(string); query != "" {
+						newRemoteIndex.Query = &query
```

Comment:
I think we want to unmarshal this string into `map[string]interface{}` here since the underlying API will be expecting a JSON object, not a plain string.

Reply:
This is configured the same way it is in for Indicies, https://github.com/mholttech/terraform-provider-elasticstack/blob/558cef1ef1e5ffa174f78971c6165832be8551ec/internal/kibana/role.go#L407-L409

Is the API expecting this to be JSON for remote_indices and string for indicies?